components/common/MainActionButton/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { MouseEvent, useCallback } from 'react';
+import { useNetwork, useSwitchNetwork } from 'wagmi';
+import { useTranslation } from 'react-i18next';
+import { LoadingButton, type LoadingButtonProps } from '@mui/lab';
+import { Button } from '@mui/material';
+
+import { useWeb3 } from 'hooks/useWeb3';
+import { useModal } from 'contexts/ModalContext';
+
+function MainActionButton({ onClick, ...props }: LoadingButtonProps) {
+  const { t } = useTranslation();
+  const { isConnected, chain: displayNetwork, connect, impersonateActive, exitImpersonate } = useWeb3();
+  const { chain } = useNetwork();
+  const { switchNetworkAsync, isLoading } = useSwitchNetwork();
+  const { close } = useModal('rollover');
```

Comment:
future improvement: to close any modal, we should be able to pass `undefined`. Although is not part of this PR change we should change this in the `useModal` hook in the future or at least add a `TODO` comment now